tutorials/prepare_dataset.md

```diff
@@ -45,9 +44,25 @@ python scripts/prepare_alpaca.py \
  --checkpoint_dir checkpoints/tiiuae/falcon-7b
 ```
 
+#### Truncating datasets
+
+By default, the finetuning script (for example [`finetuning/lora.py`](../finetuning/lora.py)) will determine the size of the longest tokenized sample in the dataset to determine the block size. However, if you are willing to truncate a few examples in the training set, you can reduce the computational resource requirements significantly. For instance you can set a sequence length threshold via `--max_seq_length`. We can determine an appropriate maximum sequence length by considering the distribution of the data sample lengths shown in the histogram below.
+
+<img src="images/prepare_dataset/alpaca.jpg" width=400px>
+
+In this case, a cut-off of 256 may be a reasonable choice:
+
+```bash
+python scripts/prepare_alpaca.py \
+ --checkpoint_dir checkpoints/tiiuae/falcon-7b \
+ --max_seq_length 256
+```
+
+For comparison, the Falcon 7B model requires xx GB of memory for the original Alpaca dataset and xx GB of memory for the truncated Alpaca dataset when finetuning with LoRA using a micro batchsize of 1 and bfloat-16 precision.
```

Comment:
LGTM! We can merge this once this is filled in

Reply:
Done!